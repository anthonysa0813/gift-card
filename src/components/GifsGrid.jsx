import React, { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

const GifsGrid = ({ cats }) => {
  const [gifs, setGifs] = useState([]);
  useEffect(() => {
    getGifs(cats).then((img) => setGifs(img));
  }, [cats]);

  return (
    <>
      {gifs.map((gif) => (
        <div
          className="container-card animate__animated animate__jackInTheBox"
          key={gif.id}
        >
          <div className="container-images">
            <img src={gif.images} alt={gif.name} />
          </div>
          <h5>{gif.name}</h5>
        </div>
      ))}
    </>
  );
};

export default GifsGrid;
