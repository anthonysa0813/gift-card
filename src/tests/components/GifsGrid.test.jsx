import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifsGrid from "../../components/GifsGrid";

describe("Pruebas en GifsGrid", () => {
  test("GifsGrid debe de estar correctamente", () => {
    const wrapper = shallow(<GifsGrid />);
    expect(wrapper).toMatchSnapshot();
  });
});
