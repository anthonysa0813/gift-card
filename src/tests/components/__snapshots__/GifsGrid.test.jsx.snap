// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifsGrid GifsGrid debe de estar correctamente 1`] = `<Fragment />`;
